CppSamples/Maps/CreateDynamicBasemapGallery/BasemapStyleListModel.cpp

```diff
@@ -51,16 +51,21 @@ QVariant BasemapStyleListModel::data(const QModelIndex& index, int role) const
 
 int BasemapStyleListModel::rowCount(const QModelIndex& parent) const
 {
+    Q_UNUSED(parent)
     return m_previews.size();
 }
 
 void BasemapStyleListModel::insertItemsIntoGallery(const QList<BasemapStyleInfo*> infos)
 {
     beginResetModel();
     m_previews.clear();
-    for (const BasemapStyleInfo* info : infos)
+    for (auto index = 0; index < infos.size(); ++index)
     {
-        m_previews.append(std::move(info));
+        m_previews.insert(index, std::move(infos.at(index)));
+        QModelIndex modelIndex = createIndex(index, 0);
+        connect(m_previews.at(index), &BasemapStyleInfo::thumbnailUrlChanged, this, [this, modelIndex]() {
+            emit dataChanged(modelIndex, modelIndex);
+        });
```

Comment:
Use consistent bracket style. Also is it possible to just capture the index initially rather than the QModelIndex?
```suggestion
        connect(m_previews.at(index), &BasemapStyleInfo::thumbnailUrlChanged, this, [this, index]() 
        {
            const QModelIndex modelIndex = createIndex(index, 0);
            emit dataChanged(modelIndex, modelIndex);
        });
```

This code may not work, proceed with caution.

Reply:
Tested and working.